perun-core/src/test/java/cz/metacentrum/perun/core/entry/MembersManagerEntryIntegrationTest.java

```diff
@@ -88,6 +88,37 @@ public void setUp() throws Exception {
 	}
 
 	@Test
+	public void findSponsoredMembers() throws Exception {
+		System.out.println(CLASS_NAME + "findSponsoredMembers");
+
+		Member sponsorMember = setUpMember3(createdVo);
+		User sponsorUser = perun.getUsersManagerBl().getUserByMember(sess, sponsorMember);
+		Group sponsors = new Group("sponsors","users able to sponsor");
+		sponsors = perun.getGroupsManagerBl().createGroup(sess,createdVo,sponsors);
+		AuthzResolverBlImpl.setRole(sess, sponsors, createdVo, Role.SPONSOR);
+		perun.getGroupsManagerBl().addMember(sess,sponsors,sponsorMember);
+
+		//no sponsored member has been created yet
+		assertTrue(perun.getMembersManagerBl().getSponsoredMembers(sess, createdVo).size() == 0);
+
+		Member sponsoredMember = perun.getMembersManagerBl().createSponsoredMember(sess, createdVo, "dummy", "Ing. Petr Draxler", "secret", sponsorUser, false);
+
+		//should contain one sponsored member
+		List<Member> sponsoredMembers = perun.getMembersManagerBl().getSponsoredMembers(sess, createdVo);
+
+		assertTrue(sponsoredMembers.size() == 1);
+		assertTrue(sponsoredMembers.get(0).equals(sponsoredMember));
```

Comment:
You can use ``list.contains(object)`` which is more safe than ``list.get(int)``. This will work also for second case, where you get two sponsored members - you can check that they are equals to create method result, not just "not equals" to each other.